### ISSUE TITLE
Potentially Insecure Evaluation of User Input in Tiled Array Slicing

### ISSUE THREAD

**Issue by hyperrealist** [bluesky/tiled#751]
Tiled uses `eval()` on user input for parsing array slices, which is not ideal.

https://github.com/bluesky/tiled/blob/9059fd0dd803e784e226d91ef76b5c2fb432a552/tiled/server/dependencies.py#L174

**Comment by danielballan** [comment]
Thanks for working on this @hyperrealist.

I want to be clear for anyone reading that there are no known exploitable vulnerabilities in the current codebase. This `regex` input validation:

https://github.com/bluesky/tiled/blob/9059fd0dd803e784e226d91ef76b5c2fb432a552/tiled/server/dependencies.py#L163

rejects malicious input. For example:

```
$ http "https://tiled-demo.blueskyproject.io/api/v1/array/full/generated/small_image?slice=import os;os.system('rm -rf /')"
HTTP/1.1 422 Unprocessable Entity
Connection: keep-alive
Content-Length: 261
Content-Type: application/json
Date: Fri, 31 May 2024 20:02:40 GMT
Server: nginx/1.18.0 (Ubuntu)
Set-Cookie: tiled_csrf=GKH9tESfusL742-9G0XY5Rn8TZvqxLQyKg1gka7-42o; HttpOnly; Path=/; SameSite=lax
server-timing: app;dur=8.2
set-cookie: tiled_csrf=GKH9tESfusL742-9G0XY5Rn8TZvqxLQyKg1gka7-42o; HttpOnly; Path=/; SameSite=lax
x-tiled-request-id: b558539f3ba5d4fa

{
    "detail": [
        {
            "ctx": {
                "pattern": "^[-0-9,:]*$"
            },
            "input": "import os;os.system('rm -rf /')",
            "loc": [
                "query",
                "slice"
            ],
            "msg": "String should match pattern '^[-0-9,:]*$'",
            "type": "string_pattern_mismatch",
            "url": "https://errors.pydantic.dev/2.7/v/string_pattern_mismatch"
        }
    ]
}
```

However, I very much support refactoring to remove `eval` for defense in depth.

**Comment by danielballan** [comment]
Closed by #752
